RUFAS/biophysical/animal/nutrients/beef_stocker_requirements_calculator.py

```
@@ -0,0 +1,195 @@
+        Raises
+        ------
+        ValueError
+            If animal_type is not a beef stocker type; if body_weight or
+            mature_body_weight are not positive and finite; or if sex is not a
+            recognised value in SEX_NEm_MULTIPLIER.
+
+        """
+        if not inputs.animal_type.is_beef_stocker:
+            raise ValueError(
+                f"BeefStockerRequirementsCalculator only handles stocker types; " f"got {inputs.animal_type.value}."
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)